### PR TITLE
Log errors

### PR DESCRIPTION
**Pull request by tomrf1**
1. Use a proper logger
2. Handle exceptions thrown by the csv library and log them. Currently one failed line can throw an exception, causing the rest of the file to be abandoned.
3. Log capi errors

TODO - add an alarm for errors once I can see, understand and stop all existing errors

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -7,13 +7,16 @@ name := "podcasts-analytics-lambda"
 
 lazy val root = (project in file(".")).enablePlugins(JavaAppPackaging, RiffRaffArtifact)
 
+val log4jVersion = "2.11.0"
+
 libraryDependencies ++= Seq(
   "org.joda" % "joda-convert" % "1.8.1",
-  "com.typesafe.scala-logging" %% "scala-logging" % "3.9.0",
+  "org.apache.logging.log4j" %% "log4j-api-scala" % "11.0",
+  "com.amazonaws" % "aws-lambda-java-log4j2" % "1.1.0",
```
Comment (regiskuckaertz) [inline build.sbt:13]: Can all the AWS dependencies share the same version number?

Comment (tomrf1) [reply build.sbt:13]: indeed, we've now get 3 different versions!

File: src/main/scala/com/gu/contentapi/utils/CsvParser.scala [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+import scala.util.{ Failure, Success, Try }
+import scala.util.control.NonFatal
+import org.apache.logging.log4j.scala.Logging
+
+class CsvParser[T: RawFieldsConverter] extends Logging {
+  private val reader = CSVReader[T]
+
+  def parseLine(line: String): Option[T] = {
+    //despite returning a List[Try[T]], readCSVFromString itself can throw an exception, so catch that as well
```
Comment (regiskuckaertz) [inline src/main/scala/com/gu/contentapi/utils/CsvParser.scala]: This is ridiculous 🤣 You might as well use the unsafe version then, and avoid having each value wrapped in a `Try`.
```scala
try {
  reader.readCSVFromString(line).headOption
} catch {
  case e if NonFatal(e) =>
    logger.warn(...)
    None
}
```

Comment (tomrf1) [reply src/main/scala/com/gu/contentapi/utils/CsvParser.scala]: It is strange. I agree - the unsafe version will throw the same exceptions that the safe version catches

File: build.sbt (hunk 1 above)
Comment (tomrf1) [reply build.sbt:13]: actually, it appears these libraries all follow their own versioning

File: build.sbt [hunk 3]
```diff
@@ -7,13 +7,16 @@ name := "podcasts-analytics-lambda"
 
 lazy val root = (project in file(".")).enablePlugins(JavaAppPackaging, RiffRaffArtifact)
 
+val log4jVersion = "2.11.0"
```
Comment (hosamaly) [inline build.sbt]: This variable seems unused.

**Review by regiskuckaertz — APPROVED** [review]

**Pull request merged**